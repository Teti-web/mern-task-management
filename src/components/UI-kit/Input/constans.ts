export const INPUT_TEST_ID = "inputComponent";
