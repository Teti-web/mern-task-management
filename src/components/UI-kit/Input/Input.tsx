import { FC } from "react";
import { INPUT_TEST_ID } from "./constans";

export interface InputType {
  type: string;
  placeholder?: string;
  isLabel: boolean;
  label?: string;
}

const Input: FC<InputType> = ({ type, placeholder, isLabel, label }) => {
  const className =
    "font-primary font-medium text-[12px] rounded-[10px] border-[1px] border-primary-100 text-secondinary-500 pt-[16px] pl-[20px] pb-[16px] pr-[20px]placeholder:text-secondinary-400 hover:border-primary-300 transition-all duration-500 ease-in-out focus:border-primary-300 active:border-primary-300 block";

  const classNameLabel =
    "font-primary text-secondinary-500 font-semibold text-[14px] leading-4 grid gap-[16px]";

  return isLabel ? (
    <label className={classNameLabel}>
      {label}
      <input
        data-testid={INPUT_TEST_ID}
        className={className}
        type={type}
        placeholder={placeholder}
      />
    </label>
  ) : (
    <input
      data-testid={INPUT_TEST_ID}
      className={className}
      type={type}
      placeholder={placeholder}
    />
  );
};

export default Input;
