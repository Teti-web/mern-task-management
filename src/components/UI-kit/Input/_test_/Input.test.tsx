import "@testing-library/jest-dom";
import { render, fireEvent, cleanup } from "@testing-library/react";

import { INPUT_TEST_ID } from "../constans";
import Input from "../Input";

describe("Input component", () => {
  afterEach(cleanup);

  it("renders input without label", () => {
    const onChangeMock = jest.fn();
    const { getByTestId, asFragment } = render(
      <Input type="text" isLabel={false} onChange={onChangeMock} />
    );
    const inputElement = getByTestId(INPUT_TEST_ID);
    expect(inputElement).toBeInTheDocument();

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders input with label", () => {
    const onChangeMock = jest.fn();
    const { getByText, getByTestId, asFragment } = render(
      <Input
        type="text"
        isLabel={true}
        label="Username"
        onChange={onChangeMock}
      />
    );
    const labelElement = getByText("Username");
    const inputElement = getByTestId(INPUT_TEST_ID);
    expect(labelElement).toBeInTheDocument();
    expect(inputElement).toBeInTheDocument();

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders input with placeholder", () => {
    const onChangeMock = jest.fn();
    const placeholderText = "Enter your username";
    const { getByPlaceholderText, asFragment } = render(
      <Input
        type="text"
        isLabel={false}
        placeholder={placeholderText}
        onChange={onChangeMock}
      />
    );
    const inputElement = getByPlaceholderText(placeholderText);
    expect(inputElement).toBeInTheDocument();

    expect(asFragment()).toMatchSnapshot();
  });

  it("calls onChange event handler", () => {
    const onChangeMock = jest.fn();
    const { getByTestId, asFragment } = render(
      <Input type="text" isLabel={false} onChange={onChangeMock} />
    );
    const inputElement = getByTestId(INPUT_TEST_ID);
    fireEvent.change(inputElement, { target: { value: "test" } });
    expect(onChangeMock).toHaveBeenCalledTimes(1);

    expect(asFragment()).toMatchSnapshot();
  });
});
