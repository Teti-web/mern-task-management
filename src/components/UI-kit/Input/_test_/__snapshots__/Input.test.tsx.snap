// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component calls onChange event handler 1`] = `
<DocumentFragment>
  <input
    class="font-primary font-medium text-[12px] rounded-[10px] border-[1px] border-primary-100 text-secondinary-500 pt-[16px] pl-[20px] pb-[16px] pr-[20px]placeholder:text-secondinary-400 hover:border-primary-300 transition-all duration-500 ease-in-out focus:border-primary-300 active:border-primary-300 block"
    data-testid="inputComponent"
    type="text"
  />
</DocumentFragment>
`;

exports[`Input component renders input with label 1`] = `
<DocumentFragment>
  <label
    class="font-primary text-secondinary-500 font-semibold text-[14px] leading-4 grid gap-[16px]"
  >
    Username
    <input
      class="font-primary font-medium text-[12px] rounded-[10px] border-[1px] border-primary-100 text-secondinary-500 pt-[16px] pl-[20px] pb-[16px] pr-[20px]placeholder:text-secondinary-400 hover:border-primary-300 transition-all duration-500 ease-in-out focus:border-primary-300 active:border-primary-300 block"
      data-testid="inputComponent"
      type="text"
    />
  </label>
</DocumentFragment>
`;

exports[`Input component renders input with placeholder 1`] = `
<DocumentFragment>
  <input
    class="font-primary font-medium text-[12px] rounded-[10px] border-[1px] border-primary-100 text-secondinary-500 pt-[16px] pl-[20px] pb-[16px] pr-[20px]placeholder:text-secondinary-400 hover:border-primary-300 transition-all duration-500 ease-in-out focus:border-primary-300 active:border-primary-300 block"
    data-testid="inputComponent"
    placeholder="Enter your username"
    type="text"
  />
</DocumentFragment>
`;

exports[`Input component renders input without label 1`] = `
<DocumentFragment>
  <input
    class="font-primary font-medium text-[12px] rounded-[10px] border-[1px] border-primary-100 text-secondinary-500 pt-[16px] pl-[20px] pb-[16px] pr-[20px]placeholder:text-secondinary-400 hover:border-primary-300 transition-all duration-500 ease-in-out focus:border-primary-300 active:border-primary-300 block"
    data-testid="inputComponent"
    type="text"
  />
</DocumentFragment>
`;
